src/SimpleDetailsList.tsx

```diff
@@ -87,16 +89,16 @@ export class SimpleDetailsList extends React.Component<any, IDetailsListBasicExa
     return (
       <div className='ms-DetailsListBasicExample'>
         <Toggle
-          isToggled={ isGridVisible }
+          checked={ isGridVisible }
           onChanged={ isToggled => this.setState({ isGridVisible: isToggled }) }
           label='Grid visible'
           onText='On'
-          offText='Off' 
+          offText='Off'
           />
 
         <CommandBar items={ this._getCommandItems() } />
 
-        <div style={ isGridVisible ? null : { display: "none" } }>
+        { isGridVisible && (

```

Comment:
I sent more explanation in mail. If you have a suggestion, let me know, but any control that is impacted by width calculations will have the same issue. There is no container resize event. You can always trigger a window resize if you need recalculations, or you can forceUpdate I believe if you have a scenario that requires forcing update. Otherwise the best solution is to just render what is necessary when it is necessary.